include/zephyr/drivers/watchdog.h

```diff
@@ -194,26 +192,26 @@ static inline int z_impl_wdt_disable(const struct device *dev)
 }
 
 /**
- * @brief Install new timeout.
+ * @brief Install a new timeout.
  *
- * This function must be used before wdt_setup(). Changes applied here
+ * @note This function must be used before wdt_setup(). Changes applied here
  * have no effects until wdt_setup() is called.
  *
- * @param dev Pointer to the device structure for the driver instance.
- * @param cfg Pointer to timeout configuration structure.
+ * @param dev Watchdog device instance.
+ * @param[in] cfg Timeout configuration.
```

Comment:
Why is `[in]` added only for this one parameter?

Reply:
Becuase it's a read-only parameter. For devices, we may modify its state, but strictly speaking, the device reference itself is a read-only parameter as well. We should probably adjust all docs with `[in]` for devices?